src/mpi/coll/ialltoall/ialltoall.c

```diff
@@ -285,17 +285,13 @@ int MPIR_Ialltoall_impl(const void *sendbuf, int sendcount,
         /* intracommunicator */
         switch (MPIR_Ialltoall_intra_algo_choice) {
             case MPIR_IALLTOALL_INTRA_ALGO_GENTRAN_RING:
-                mpi_errno =
+                return
```

Comment:
From what I can tell, returning from the switch statement is not common across the MPICH collective code. The typical pattern I see is

> 		if (mpi_errno)
>                     MPIR_ERR_POP(mpi_errno);
> 		goto fn_exit;
> 		break;
> 

I think we shouldn't break the convention unless we plan to modify all the collectives.

Reply:
ok